dffml/model/model.py

```diff
@@ -84,8 +84,14 @@ def __init__(self, config):
         # TODO Just in case its a string. We should make it so that on
         # instantiation of an @config we convert properties to their correct
         # types.
-        if isinstance(getattr(self.config, "directory", None), str):
+        directory = getattr(self.config, "directory", None)
+        if isinstance(directory, str):
             self.config.directory = pathlib.Path(self.config.directory)
+            if isinstance(directory, pathlib.Path):
```

Comment:
Almost there! This `if isinstance(directory, pathlib.Path):` and it's `self.config = ...` block need to be indented one to the left. Because currently we'll miss this block unless it was a `str` to being with.

Reply:
thanks for the help :). I have altered that in the latest comment